tests/tarray_ut/tarray_ut.c

```diff
@@ -196,6 +201,131 @@ TEST_FUNCTION(TARRAY_of_THANDLE_can_be_created)
+    STRICT_EXPECTED_CALL(malloc_2(N_ELEMENTS, sizeof(uint32_t))); /*inner array*/
+
+    ///act
+    TARRAY(uint32_t) arr_of_uint32_t = TARRAY_CREATE_WITH_CAPACITY(uint32_t)(N_ELEMENTS);
+
+    ///assert
+    ASSERT_IS_NOT_NULL(arr_of_uint32_t);
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+
+    for (uint32_t i = 0; i < N_ELEMENTS; i++)
+    {
+        arr_of_uint32_t->arr[i] = 4; /*assignment is possible*/
+    }
+
+    // cleanup
+    TARRAY_ASSIGN(uint32_t)(&arr_of_uint32_t, NULL);
+}
+
+/* Tests_SRS_TARRAY_01_005: [ If there are any failures then TARRAY_CREATE_WITH_CAPACITY(T) shall fail and return NULL. ]*/
+TEST_FUNCTION(when_underlying_calls_fail_TARRAY_CREATE_WITH_CAPACITY_also_fails)
+{
+    ///arrange
+    STRICT_EXPECTED_CALL(malloc(IGNORED_ARG)); /*struct*/
+    STRICT_EXPECTED_CALL(malloc_2(1, sizeof(uint32_t))); /*inner array*/
+
+    umock_c_negative_tests_snapshot();
+
+    for (size_t i = 0; i < umock_c_negative_tests_call_count(); i++)
+    {
+        if (umock_c_negative_tests_can_call_fail(i))
+        {
+            umock_c_negative_tests_reset();
+            umock_c_negative_tests_fail_call(i);
+
+            ///act
+            TARRAY(uint32_t) arr_of_uint32_t = TARRAY_CREATE_WITH_CAPACITY(uint32_t)(1);
+
+            // assert
+            ASSERT_IS_NULL(arr_of_uint32_t, "On failed call %zu", i);
+        }
+    }
+}
+
+#define N_ELEMENTS 1000
+
+/* Tests_SRS_TARRAY_01_002: [ TARRAY_CREATE_WITH_CAPACITY(T) shall call THANDLE_MALLOC to allocate the result. ]*/
+/* Tests_SRS_TARRAY_01_003: [ TARRAY_CREATE_WITH_CAPACITY(T) shall call malloc_2 to allocate capacity entries for result->arr. ]*/
+/* Tests_SRS_TARRAY_01_004: [ TARRAY_CREATE_WITH_CAPACITY(T) shall succeed and return a non-NULL value. ]*/
+TEST_FUNCTION(MU_C3(TARRAY_CREATE_WITH_CAPACITY_with_, N_ELEMENTS, _THANDLE_A_TEST_TARRAY_succeeds))
+{
+    ///arrange
+
+    /*array element management is entirely in user's hands. TARRAY only knows about ensuring capacity for an array.*/
+    THANDLE(A_TEST) a = THANDLE_MALLOC(A_TEST)(NULL); /*NULL = no special handling of the int inside the A_TEST*/
+    ASSERT_IS_NOT_NULL(a);
+    umock_c_reset_all_calls();
+
+    STRICT_EXPECTED_CALL(malloc(IGNORED_ARG)); /*struct*/
+    STRICT_EXPECTED_CALL(malloc_2(N_ELEMENTS, sizeof(THANDLE(A_TEST)))); /*inner array*/
+
+    ///act
+    TARRAY(THANDLE(A_TEST)) arr_of_a = TARRAY_CREATE_WITH_CAPACITY(THANDLE(A_TEST))(N_ELEMENTS);
+
+    ///assert
+    ASSERT_IS_NOT_NULL(arr_of_a);
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+
+    for (uint32_t i = 0; i < N_ELEMENTS; i++)
+    {
+        THANDLE_INITIALIZE(A_TEST)(&arr_of_a->arr[i], a);
+        THANDLE_ASSIGN(A_TEST)(&arr_of_a->arr[i], NULL);
+        THANDLE_ASSIGN(A_TEST)(&a, NULL);
+    }
+
+    // cleanup
+    TARRAY_ASSIGN(THANDLE(A_TEST))(&arr_of_a, NULL);
+}
+
+/* TARRAY_ENSURE_CAPACITY(T) */
```

Comment:
>TARRAY_ENSURE_CAPACITY [](http://example.com/codeflow?start=3&length=22)

consider adding at least 1 test for an array that is created with initial capacity and AFTER it is resized.

"should not fail".